### PR TITLE
[Improvement-16106] Add admin-user-filter in LDAP

### LINKED ISSUE #16106: [Improvement][dolphinscheduler-api] Make ADMIN user a "memberOf" when using LDAP server
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

The currently implementation will only llow one user to be ADMIN for a scheduler instance using the attribute

security.authentication.ldap.user.admin=read-only-admin

Further more the revoke/invoking must be done changing the dolphin config. In large organisation this will done through roles/groups.

Therefore, I would like to be assign admin access using LDAP groups ("memberOf" attribute)

So can we extend the code (keeping the old way too), but also add chacke like "mail" attributes

        security.authentication.ldap.user.admin-attribute=membmerOf
        security.authentication.ldap.user.admin-value=cn=dolphinAdmin,dc=....etc...

https://github.com/apache/dolphinscheduler/pull/16105

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SbloodyS**
## Purpose of the pull request

close #16106
close #16105

## Brief change log

- Add admin-user-filter in LDAP
- Add api-test for LDAP login
- LDAP docs improvement

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/LdapLoginResult.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class LdapLoginResult {
+
+    boolean success;
+    String ldapEmail;
+    UserType userType;
+    String userName;
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/LdapLoginResult.java]:
```suggestion
    private boolean success;
    private String ldapEmail;
    private UserType userType;
    private String userName;
```

**Review by ruanwenjun — DISMISSED** [review]
LGTM

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/LdapLoginResult.java (hunk 1 above)
Comment (SbloodyS) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/LdapLoginResult.java]: Fixed.

**Review by ruanwenjun — DISMISSED** [review]

**Review by ruanwenjun — APPROVED** [review]

**Review by caishunfeng — APPROVED** [review]
LGTM

**Pull request merged**